lib/dns/zone.js

```diff
@@ -0,0 +1,817 @@
+ * add to your zone, or to refer to an existing one.
+ *
+ * Note that using this method will not itself make any API requests. You will
+ * use the object returned in other API calls, for example to add a record to
+ * your zone or to delete an existing one.
+ *
+ * @param {string} type - The type of record to construct or the type of record
+ *     you are referencing.
+ * @param {object} metadata - The metadata of this record.
+ * @param {string} metadata.name - The name of the record, e.g.
+ *     `www.example.com.`.
+ * @param {string[]} metadata.data - Defined in
+ *     [RFC 1035, section 5](https://goo.gl/9EiM0e) and
+ *     [RFC 1034, section 3.6.1](https://goo.gl/Hwhsu9).
+ * @param {number} metadata.ttl - Seconds that the resource is cached by
+ *     resolvers.
+ * @return {module:dns/record}
+ *
+ * @example
+ * //-
+ * // Reference an existing record to delete from your zone.
+ * //-
+ * var oldARecord = zone.record('a', {
+ *   name: 'example.com.',
+ *   data: '1.2.3.4',
+ *   ttl: 86400
+ * });
+ *
+ * //-
+ * // Construct a record to add to your zone.
+ * //-
+ * var newARecord = zone.record('a', {
+ *   name: 'example.com.',
+ *   data: '5.6.7.8',
+ *   ttl: 86400
+ * });
+ *
+ * //-
+ * // Use these records together to create a change.
+ * //-
+ * zone.createChange({
+ *   add: newARecord,
+ *   delete: oldARecord
+ * }, function(err, change, apiResponse) {});
+ */
+Zone.prototype.record = function(type, metadata) {
+  return new Record(this, type, metadata);
+};
+
+/**
+ * Provide a record type that should be deleted and replaced with other records.
+ *
+ * <strong>This is not an atomic request.</strong> Two API requests are made
+ * (one to get records of the type that you've requested, then another to
+ * replace them), which means the operation is not atomic and could result in
+ * unexpected changes.
+ *
+ * @param {string|string[]} recordTypes - Type(s) of records to replace.
+ * @param {module:dns/record|module:dns/record[]} newRecords - The record
+ *     objects to add.
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An API error.
+ * @param {?module:dns/change} callback.change - A {module:dns/change} object.
+ * @param {?object} callback.apiResponse - Raw API response.
+ *
+ * @example
+ * var newNs1Record = zone.record('ns', {
+ *   name: 'example.com.',
+ *   data: 'ns-cloud1.googledomains.com.',
+ *   ttl: 86400
+ * });
+ *
+ * var newNs2Record = zone.record('ns', {
+ *   name: 'example.com.',
+ *   data: 'ns-cloud2.googledomains.com.',
+ *   ttl: 86400
+ * });
+ *
+ * var newNsRecords = [
+ *   newNs1Record,
+ *   newNs2Record
+ * ];
+ *
+ * zone.replaceRecords('ns', newNsRecords, function(err, change, apiResponse) {
+ *   if (!err) {
+ *     // The change was created successfully.
+ *   }
+ * });
+ */
+Zone.prototype.replaceRecords = function(recordType, newRecords, callback) {
+  var self = this;
+
+  this.getRecords(recordType, function(err, recordsToDelete) {
```

Comment:
When we use `getRecords`, it will actually run it until no more results exist. That's new as of a couple merges ago when we introduced streamRouter across the library. The bottom of this file uses "streamRouter.extend()" to list the methods that should be wrapped.